src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -238,218 +249,237 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, containerDAO:
+            }
+            (entityUpdate, trial)
           }
-          val trial = Try {
-            val updatedEntity = applyOperationsToEntity(entity, entityUpdate.operations)
-            containerDAO.entityDAO.save(workspaceContext, updatedEntity, txn)
+          val errorMessages = results.collect {
+            case (entityUpdate, Failure(regrets)) => s"Could not update ${entityUpdate.entityType} ${entityUpdate.name} : ${regrets.getMessage}"
+          }
+          if (errorMessages.isEmpty) {
+            RequestComplete(StatusCodes.NoContent)
+          } else {
+            txn.setRollbackOnly()
+            RequestComplete(StatusCodes.BadRequest, errorMessages)
           }
-          (entityUpdate, trial)
-        }
-        val errorMessages = results.collect {
-          case (entityUpdate, Failure(regrets)) => s"Could not update ${entityUpdate.entityType} ${entityUpdate.name} : ${regrets.getMessage}"
-        }
-        if (errorMessages.isEmpty) {
-          RequestComplete(StatusCodes.NoContent)
-        } else {
-          txn.setRollbackOnly()
-          RequestComplete(StatusCodes.BadRequest, errorMessages)
         }
       }
     }
 
-  def listEntityTypes(workspaceName: WorkspaceName): PerRequestMessage =
-    dataSource inTransaction { txn =>
+  def listEntityTypes(workspaceName: WorkspaceName): Future[PerRequestMessage] =
+    dataSource inFutureTransaction { txn =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevel.Read, txn) { workspaceContext =>
-        RequestComplete(containerDAO.entityDAO.getEntityTypes(workspaceContext, txn).toSeq)
+        Future.successful(RequestComplete(containerDAO.entityDAO.getEntityTypes(workspaceContext, txn)))
       }
     }
 
-  def listEntities(workspaceName: WorkspaceName, entityType: String): PerRequestMessage =
-    dataSource inTransaction { txn =>
+  def listEntities(workspaceName: WorkspaceName, entityType: String): Future[PerRequestMessage] =
+    dataSource inFutureTransaction { txn =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevel.Read, txn) { workspaceContext =>
-        RequestComplete(containerDAO.entityDAO.list(workspaceContext, entityType, txn).toList)
+        Future.successful(RequestComplete(containerDAO.entityDAO.list(workspaceContext, entityType, txn).toList))
       }
     }
 
-  def getEntity(workspaceName: WorkspaceName, entityType: String, entityName: String): PerRequestMessage =
-    dataSource inTransaction { txn =>
+  def getEntity(workspaceName: WorkspaceName, entityType: String, entityName: String): Future[PerRequestMessage] =
+    dataSource inFutureTransaction { txn =>
       withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevel.Read, txn) { workspaceContext =>
         withEntity(workspaceContext, entityType, entityName, txn) { entity =>
-          PerRequest.RequestComplete(entity)
+          Future.successful(PerRequest.RequestComplete(entity))
```

Comment:
because sometimes we're "scheduling" future work, and other times (like here) we're just returning a completed Future:  in this case we've done all the work and just need to return a completed Future[PerRequestMessage].